### ISSUE TITLE
Cryptic error message when assigning to a key also present on the constructor, but forgetting to declare it in the class body

### ISSUE THREAD

**Issue by KilianKilmister** [microsoft/TypeScript#41227]
**TypeScript Version:**  4.1.0-dev.20201024

**Search Terms:**
static, member, method, property

### Scenario

```ts
class MyClass {
  static someKey: 'someValue'
  constructor () {
    // intentionally using the same key as on the constructor, but forgetting to declare it in the class body.
    this.someKey = new.target.someKey
  }
}
```
**Expected behavior:**

Error pointing to the issue
`Error => <prop> doesn't exist on type <Type>`

**Actual behavior:**

Crypting error, that's stating the obvious, but not stating an actual reason for the error
`Property <Prop> is a static member of type <Type>`

**Related Issues:**

**Code**
```ts
class MyClass<T> {
  constructor (arg: T) {
    // cryptic error
    // err > Property 'someMethod' is a static member of type 'MyClass<T>'.
    this.someMethod = new.target.someMethod.bind(new.target, arg)
    // clear error
    // err > Property 'key' does not exist on type 'MyClass<T>'.
    this.key = 'whatever'
  }
  static someMethod <T>(arg: T) {
    console.log(arg)
  }
}

class MyFunctionalClass<T> {
  someMethod: () => void
  key: string
  constructor (arg: T) {
    this.someMethod = new.target.someMethod.bind(new.target, arg)
    this.key = 'whatever'
  }
  static someMethod <T>(arg: T) {
    console.log(arg)
  }
}
```

<details><summary><b>Output</b></summary>

```ts
"use strict";
class MyClass {
    constructor(arg) {
        // cryptic error
        // err > Property 'someMethod' is a static member of type 'MyClass<T>'.
        this.someMethod = new.target.someMethod.bind(new.target, arg);
        // clear error
        // err > Property 'key' does not exist on type 'MyClass<T>'.
        this.key = 'whatever';
    }
    static someMethod(arg) {
        console.log(arg);
    }
}
class MyFunctionalClass {
    someMethod;
    key;
    constructor(arg) {
        this.someMethod = new.target.someMethod.bind(new.target, arg);
        this.key = 'whatever';
    }
    static someMethod(arg) {
        console.log(arg);
    }
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": true,
    "alwaysStrict": true,
    "noEmitHelpers": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "esModuleInterop": true,
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": 2,
    "target": "Latest",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play?strictPropertyInitialization=false&useDefineForClassFields=true&noEmitHelpers=true&noUnusedLocals=true&noUnusedParameters=true&target=99&jsx=0&ts=4.1.0-beta#code/MYGwhgzhAECyCeBhcUA8AVAfNA3gKGmmAHsA7CAFwCcBXYC4q6ACjCoHMAuadASlwKFoAemFEq8AA4UAlsGgBTKlUaDCoxcujYACislKK8aAHIIxALYLYCigAtiAExPQZMMNEphZ8qxYBGStDEAGbQRgamCMiQEBiYJgB0auF2bonmVjb2TtAAvNCkCgDuiRRs7LYZlta2Do6J-jKkjsxFpeUctgA00BW8KRqgCmyaKlSDYkpMuvqGxiYA1grwLo7ECjCkxBSKAB5uu2ThUgpRSChxWEkp9unLxgUmxXbeCgBuSiaCAL6CXj5PDVsvVoPFWBxuHwBEIiGRzCAFIkQMR2BD2ANCH8-nhQLE4PAAGI0Uj0GRkMAgGJoLAwoFZOpObjMfh5bDvYgyRyCB7cShUZrsQQkcjUOgMJjoqH8fCwu4QaoMnKOfKFEplCpVTK1ZWNZqtdoaroUXr9W5pBUPVXPV4UD5fX7-cqA7Ug3LgirSumEEUIpEotFmrF4H5AA)

**Comment by a-tarasyuk** [comment]
@RyanCavanaugh Should TS avoid checking existing static properties only in assignments?

```ts
class MyClass {
    static foo: 'foo';
    constructor () {
        this.foo = new.target.foo; // Property 'foo' does not exist on type 'MyClass'
    }
}
```

**Comment by RyanCavanaugh** [comment]
@a-tarasyuk not sure I understand the question. I think the defect is only that the error message is a bit unclear

**Comment by a-tarasyuk** [comment]
@RyanCavanaugh TS uses this message if a type has static property. Do we need to change the message?
```ts
if (typeHasStaticProperty(propNode.escapedText, containingType)) {
    errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_is_a_static_member_of_type_1, declarationNameToString(propNode), typeToString(containingType));
}
```

**Comment by RyanCavanaugh** [comment]
I think ideally we just rephrase it to something like

 > Property '0' does not exist on type '1'. Did you mean to access the static member '1'.'0' instead?

cc @DanielRosenwasser

**Comment by DanielRosenwasser** [comment]
Thanks @a-tarasyuk!
